server/pokemon/toCard.ts

```diff
@@ -0,0 +1,33 @@
+import { Card, STAT_NAME } from "../card/card.types";
+import { PokemonDto } from "./pokemon.types";
+
+function toCard(dto: PokemonDto) {
+  const stats = STAT_NAME.map((statName, index) => {
+    const value =
+      statName in dto
+        ? // @ts-ignore
```

Comment:
what does this do?

Reply:
Should have mentioned that, Jenny.  it's from a hell of a lot of Googling !

The toCard() function looks for stat (STAT_NAME) values in the Pokemon's stats AND in the main object too (weight, height and hp).  

To check a property is on the object you do

if (property_name in obj) value = obj["property_name"];

Even though you've done the check, Typescript still can't handle the truth, so you have to tell it to ignore the check.

Reply:
ahh okay thanks for the explanation!